### PR TITLE
[Request access] Add capturing of decision maker admin user id

### PR DESCRIPTION
**Pull request by erzhtor**
Follow up on https://github.com/sourcegraph/sourcegraph/pull/47741. Closes https://github.com/sourcegraph/sourcegraph/issues/47445.

This PR adds a new `access_requests.decision_by_user_id` column to store admin user id who rejected/approved access request for future usage in case needed.

## Test plan
- Tests updated; CI green
- Check diff as no visual changes
- Optionally go through submitting access request and approving/rejecting it as admin

### CODE REVIEW

File: internal/database/access_requests.go [hunk 1]
```diff
@@ -126,6 +126,7 @@ var (
 		sqlf.Sprintf("email"),
 		sqlf.Sprintf("status"),
 		sqlf.Sprintf("additional_info"),
+		sqlf.Sprintf("decision_by"),
```
Comment (thenamankumar) [inline internal/database/access_requests.go:129]: `decision_by_user_id` is a better column name IMO.

**Review by thenamankumar — APPROVED** [review]
LGTM

`DecisionByUserID` and `decision_by_user_id` are better names IMO.

File: migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE IF EXISTS access_requests DROP decision_by IF EXISTS col1;
```
Comment (sashaostrikov) [inline migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql:1]: Question: what does `col1` stand for here?

File: internal/database/access_requests.go (hunk 1 above)
Comment (sashaostrikov) [reply internal/database/access_requests.go:129]: +1

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql (hunk 2 above)
Comment (erzhtor) [reply migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql]: Thank you for catching this. This is some draft leftover, fixed b119aff98780652c4a44e41b8e2593095a2c5810.

File: internal/database/access_requests.go (hunk 1 above)
Comment (erzhtor) [reply internal/database/access_requests.go]: Good point. Renamed 4f89d5a98af841d56b5fddc51d4fbd4540e0c1fc.

File: migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql (hunk 2 above)
Comment (sashaostrikov) [reply migrations/frontend/1677700103_add_access_requests_decision_by_column/down.sql]: Nice!
You just need to run `sg generate` now to update db schema definitions.

**Pull request merged**